simulator/sender.cc

```diff
@@ -238,49 +241,55 @@ void* Sender::MasterThread(int /*unused*/) {
         }
         // Add inter-symbol delay
         DelayTicks(tick_start, GetTicksForFrame(ctag.frame_id_) * symbol_delay);
-        tick_start = GetTime::Rdtsc();
+        tick_start += (GetTicksForFrame(ctag.frame_id_) * symbol_delay);
 
         size_t next_frame_id = ctag.frame_id_;
         // Check to see if the current frame is finished
         assert(next_symbol_id <= cfg_->Frame().NumTotalSyms());
         if (next_symbol_id == cfg_->Frame().NumTotalSyms()) {
+          // Set the end time to time the last symbol was scheduled
+          // to get entire frame length it will be necessary to add another
+          // symbol time or take start(frame+1) - start(frame)
+          frame_end_us = timestamp_us;
           next_frame_id++;
 
           // Find start symbol of next frame and add proper delay
           next_symbol_id = FindNextSymbol(0);
           if ((kDebugSenderReceiver == true) ||
               (kDebugPrintPerFrameDone == true)) {
-            double timeus_now = GetTime::GetTimeUs();
             std::printf(
                 "Sender: Tx frame %d in %.2f ms, next frame %zu, start symbol "
                 "%zu\n",
-                ctag.frame_id_, (timeus_now - start_time) / 1000.0,
+                ctag.frame_id_, (frame_end_us - frame_start_us) / 1000.0,
                 next_frame_id, next_symbol_id);
-            start_time = timeus_now;
           }
+          // Set end of frame time to the time after the last symbol
+          this->frame_end_[(ctag.frame_id_ % kNumStatsFrames)] = frame_end_us;
 
           if (next_frame_id == cfg_->FramesToTest()) {
             keep_running.store(false);
             break; /* Finished */
           } else {
-            // Set end of frame time to the time after the last symbol
-            this->frame_end_[(ctag.frame_id_ % kNumStatsFrames)] = tick_start;
-
             // Wait for the inter-frame delay
             DelayTicks(tick_start, ticks_inter_frame_);
-
-            tick_start = GetTime::GetTimeUs();
+            tick_start += ticks_inter_frame_;
+            frame_start_us = GetTime::GetTimeUs();
 
             // Set the frame start time to the start time of the frame
             // (independant of symbol type)
-            this->frame_start_[(next_frame_id % kNumStatsFrames)] = tick_start;
+            this->frame_start_[(next_frame_id % kNumStatsFrames)] =
+                frame_start_us;
 
             // Wait until the first tx symbol
             DelayTicks(tick_start,
                        (GetTicksForFrame(ctag.frame_id_) * next_symbol_id));
+            tick_start += (GetTicksForFrame(ctag.frame_id_) * next_symbol_id);
           }
         }  // if (next_symbol_id == cfg_->Frame().NumTotalSyms()) {
-        tick_start = GetTime::Rdtsc();
+        else {
+          // Save schedule time of this symbol
+          timestamp_us = GetTime::GetTimeUs();
```

Comment:
I don't think there's a need to use this variable. It adds overhead and we don't really need the timestamp for each symbol.